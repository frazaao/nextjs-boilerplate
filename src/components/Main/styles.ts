import styled from 'styled-components';

export const Wrapper = styled.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const Logo = styled.img`
  padding: 2rem;
  width: 25rem;
  margin-bottom: 2rem;
  animation: rotateAnimation 15s linear infinite;

  @keyframes rotateAnimation {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
`;

export const Title = styled.h1`
  font-size: 4rem;
  margin-bottom: 2rem;
`;

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: normal;
`;
