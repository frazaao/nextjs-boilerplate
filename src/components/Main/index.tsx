import * as S from './styles';

export default function Main() {
  return (
    <S.Wrapper>
      <S.Logo src="img/reactIcon.svg" alt="Imagem de Ã¡tomo" />
      <S.Title>NextJS Boilerplate</S.Title>
      <S.Description>
        Typescript, ReactJS, NextJS and StyledComponents
      </S.Description>
    </S.Wrapper>
  );
}
