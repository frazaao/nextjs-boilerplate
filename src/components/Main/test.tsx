import { render, screen } from '@testing-library/react';
import Main from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />);
    expect(screen.getByText('NextJS Boilerplate')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the logo', () => {
    render(<Main />);
    expect(screen.getByAltText('Imagem de Ã¡tomo')).toBeInTheDocument();
  });

  it('should render the color correctly', () => {
    const { container } = render(<Main />);
    expect(container.firstChild).toHaveStyle({
      backgroundColor: '#06092b'
    });
  });
});
