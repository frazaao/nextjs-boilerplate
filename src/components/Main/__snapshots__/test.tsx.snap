// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
  -webkit-animation: rotateAnimation 15s linear infinite;
  animation: rotateAnimation 15s linear infinite;
}

.c2 {
  font-size: 4rem;
  margin-bottom: 2rem;
}

.c3 {
  font-size: 2rem;
  font-weight: normal;
}

<main
  class="c0"
>
  <img
    alt="Imagem de Ã¡tomo"
    class="c1"
    src="img/reactIcon.svg"
  />
  <h1
    class="c2"
  >
    NextJS Boilerplate
  </h1>
  <h2
    class="c3"
  >
    Typescript, ReactJS, NextJS and StyledComponents
  </h2>
</main>
`;
